### DISCUSSION TITLE
Fetching software that does not come in a tarball

### DISCUSSION THREAD

**Discussion by adamjstewart** [spack/spack#2801] [General]
Some software like [STREAM](https://www.cs.virginia.edu/stream/FTP/Code/) and [ASTRA](http://www.desy.de/~mpyflo/Astra_for_64_Bit_Linux/) does not offer a tarball for download. You need to download a directory of files from their webpage. Spack currently has no way of handling this.

**Comment by tgamblin** [comment]
@adamjstewart: can this be done with resources?

**Comment by adamjstewart** [comment]
I'm not sure if that will work. Do resources support `expand=False`?

**Comment by tgamblin** [comment]
They should support all the same parameters fetch strategies do.
